FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift

```diff
@@ -59,19 +58,19 @@ class ModelDownloadTask: NSObject {
     self.downloader = downloader
     self.telemetryLogger = telemetryLogger
     self.defaults = defaults
-    downloadStatus = .notStarted
   }
 }
 
 extension ModelDownloadTask {
   /// Name for model file stored on device.
   var downloadedModelFileName: String {
-    return "fbml_model__\(appName)__\(remoteModelInfo.name)"
+    return "fbml_model__\(appName)__\(remoteModelInfo.name).tflite"
   }
 
   func download(progressHandler: ProgressHandler?, completion: @escaping Completion) {
     /// Prevent multiple concurrent downloads.
-    guard downloadStatus == .notStarted else {
+    // TODO: Merge if there are multiple same requests.
```

Comment:
I think merging should happen on the class user level.

Reply:
Do you mean at the `ModelDownloader` level?

Reply:
yes

Reply:
Moved the TODO, thanks!